### DISCUSSION TITLE
Compor SPA com múltiplos apps ou multi-page de vários SPAs?

### DISCUSSION THREAD

**Discussion by rafaelpaixao** [frontendbr/forum#1702] [Perguntas e Respostas]
Alguém já passou ou vive essa situação onde cada time precisa de autonomia para desenvolver o "seu" app de forma independente, mas no final todos os apps precisam ser entregues como parte de um sistema único?

Já temos uma lib base que fornece o esqueleto do layout e outras coisas comuns. Utilizamos Vue JS em todos os apps e cada app é escrito como um plugin do Vue JS. De modo que existem projetos que servem apenas para agrupar os apps, de acordo com o cenário do deploy.

O problema é que esse método não é compatível com o fluxo de trabalho.
Todos os apps estão limitados pelas versões de dependências do app mais antigo.
Como cada app representa um módulo e é mantido por um time diferente, o custo de atualizar dependências em comum não entra em nenhum projeto.

Disso veio a ideia de entregar um gateway com múltiplos apps (compartilhando token e outras coisas por local storage).

Também estamos procurando consultoria com experiência em ERPs e ambientes on-premise para revisar essa arquitetura, então aceito indicações!

**Comment by felipefialho** [comment]
Acho que linka com o assunto de [Micro Frontends](#661)

**Comment by LuanDc** [comment]
De uma olhada nesse [link](https://github.com/single-spa/single-spa). Microserviços permitem criar front-end unindo micro-aplicações desacopladas. É possível, inclusive, usar múltiplos frameworks.
